contracts/exchange-wrappers/TestExchangeWrapper.sol

```diff
@@ -0,0 +1,150 @@
+ * @author dYdX
+ *
+ * An ExchangeWrapper for testing
+ */
+contract TestExchangeWrapper is
+    ExchangeWrapper
+{
+    using TokenInteract for address;
+
+    // ============ Constants ============
+
+    // arbitrary address to send the tokens to (they are burned)
+    address constant EXCHANGE_ADDRESS = address(0x1);
+
+    // ============ Structs ============
+
+    struct Order {
+        address originator;
+        address makerToken;
+        address takerToken;
+        uint256 makerAmount;
+        uint256 takerAmount;
+    }
+
+    // ============ ExchangeWrapper functions ============
+
+    function exchange(
+        address tradeOriginator,
+        address receiver,
+        address makerToken,
+        address takerToken,
+        uint256 requestedFillAmount,
+        bytes calldata orderData
+    )
+        external
+        returns (uint256)
+    {
+        Order memory order = parseData(orderData);
+
+        require(
+            order.originator == tradeOriginator,
+            "Originator mismatch"
+        );
+        require(
+            order.makerToken == makerToken,
+            "MakerToken mismatch"
+        );
+        require(
+            order.takerToken == takerToken,
+            "TakerToken mismatch"
+        );
+
+        takerToken.transfer(EXCHANGE_ADDRESS, requestedFillAmount);
+        makerToken.approve(receiver, order.makerAmount);
+
+        return order.makerAmount;
+    }
+
+    function getExchangeCost(
+        address makerToken,
+        address takerToken,
+        uint256 desiredMakerToken,
+        bytes calldata orderData
+    )
+        external
+        view
+        returns (uint256)
+    {
+        Order memory order = parseData(orderData);
+
+        require(
+            order.makerAmount == desiredMakerToken,
+            "MakerAmount mismatch"
+        );
+        require(
+            order.makerToken == makerToken,
+            "MakerToken mismatch"
+        );
+        require(
+            order.takerToken == takerToken,
+            "TakerToken mismatch"
+        );
+
+        return order.takerAmount;
+    }
+
+    // ============ Private functions ============
+
+    function parseData(
+        bytes memory data
+    )
+        private
+        pure
+        returns (Order memory)
+    {
+        require(
+            data.length == 160,
+            "Exchange data invalid length"
+        );
+
+        uint256 makerToken;
+        uint256 takerToken;
+        uint256 originator;
+        uint256 makerAmount;
+        uint256 takerAmount;
+
+        /* solium-disable-next-line security/no-inline-assembly */
+        assembly {
+            originator := mload(add(data, 32))
+            makerToken := mload(add(data, 64))
+            takerToken := mload(add(data, 92))
+            makerAmount := mload(add(data, 128))
+            takerAmount := mload(add(data, 160))
+        }
+
+        return Order({
+            originator: address(originator),
```

Comment:
why are you declaring these as `uint256`'s and then casting them? Wouldn't it work if you just declared them as addresses?